chainer/training/extensions/value_observation.py

```diff
@@ -0,0 +1,46 @@
+import time
+
+from chainer.training import extension
+
+
+def observe_value(key, target_func):
+    """Returns a trainer extension to continuously record a value.
+
+    Args:
+        key (str): Key of observation to record.
+        target_func (function): Function that returns the value to record.
+            It must take one argument: :class:~chainer.training.Trainer object.
+    Returns:
+        The extension function.
+    """
+    @extension.make_extension(
+        trigger=(1, 'epoch'), priority=extension.PRIORITY_WRITER)
+    def _observe_value(trainer):
+        trainer.observation[key] = target_func(trainer)
```

Comment:
Can we support a case that `target_func` returns multiple values (possibly with additional keys)? It can be a different extension.